### ISSUE TITLE
MAINT, TST: Python 3.9 32-bit Linux test failures

### ISSUE THREAD

**Issue by tylerjereddy** [scipy/scipy#12931]
The [PR](https://github.com/MacPython/scipy-wheels/pull/98) to add Python 3.9 entries for Travis CI (non-Windows) wheel builds is actually looking pretty solid so far. The only real (not related to availability) failure I see so far is for 32-bit Linux + Python 3.9 in this log: https://travis-ci.com/github/MacPython/scipy-wheels/jobs/396653262

Note that I'm not necessarily saying that Python 3.9 itself is the source of the failure; indeed it may be a fair bit more likely that using i.e., a newer version of a dependency like NumPy is involved in these test tunings.

Summary below the fold:

<details>

```
=================================== FAILURES ===================================
____________________________ TestCheby1.test_basic _____________________________
[gw1] linux -- Python 3.9.0 /venv/bin/python
/venv/lib/python3.9/site-packages/scipy/signal/tests/test_filter_design.py:2595: in test_basic
    assert_array_almost_equal(a, [
E   AssertionError: 
E   Arrays are not almost equal to 14 decimals
E   
E   Mismatched elements: 1 / 9 (11.1%)
E   Max absolute difference: 2.13162821e-14
E   Max relative difference: 3.36530683e-16
E    x: array([  1.              ,  -7.78240203502796,  26.54354569747454,
E          -51.82182531666388,  63.34127355102686, -49.63358186631158,
E           24.3486218294939 ,  -6.83692534860468,   0.84129349444491])
E    y: array([  1.              ,  -7.78240203502796,  26.54354569747454,
E          -51.82182531666387,  63.34127355102684, -49.63358186631157,
E           24.34862182949389,  -6.83692534860468,   0.84129349444491])
        N          = 24
        a          = array([  1.        ,  -7.78240204,  26.5435457 , -51.82182532,
        63.34127355, -49.63358187,  24.34862183,  -6.83692535,
         0.84129349])
        b          = array([2.15073272e-11, 1.72058618e-10, 6.02205162e-10, 1.20441032e-09,
       1.50551290e-09, 1.20441032e-09, 6.02205162e-10, 1.72058618e-10,
       2.15073272e-11])
        k          = 0.6384835658206038
        p          = array([0.99938388-0.03141819j, 0.9991114 -0.03195271j,
       0.99878499-0.03307606j, 0.9983673 -0.03489287j,
       0...759099j,
       0.9983673 +0.03489287j, 0.99878499+0.03307606j,
       0.9991114 +0.03195271j, 0.99938388+0.03141819j])
        self       = <scipy.signal.tests.test_filter_design.TestCheby1 object at 0xe349a2e0>
        wn         = 0.01
        z          = array([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1.])
_____________________ test_solve_generalized_discrete_are ______________________
[gw0] linux -- Python 3.9.0 /venv/bin/python
/venv/lib/python3.9/site-packages/scipy/linalg/tests/test_solvers.py:654: in test_solve_generalized_discrete_are
    _test_factory(case, min_decimal[ind])
        _test_factory = <function test_solve_generalized_discrete_are.<locals>._test_factory at 0xe27c5610>
        case       = (array([[ 0.97905961,  0.15428665, -0.03994193,  0.00391164, -0.01180311,
        -0.02494093, -0.00892079,  0.0143359...891e-06,  2.94712339e-08],
       [ 2.40952333e-08,  4.30614350e-07],
       [ 1.23662598e-08, -6.06921957e-07]]), ...)
        cases      = [(array([[0.276923  , 0.8234578 , 0.950222  ],
       [0.04617139, 0.6948286 , 0.03444608],
       [0.09713178, 0.3170...91e-06,  2.94712339e-08],
       [ 2.40952333e-08,  4.30614350e-07],
       [ 1.23662598e-08, -6.06921957e-07]]), ...)]
        ind        = 2
        mat20170120 = {'A': array([[ 0.97905961,  0.15428665, -0.03994193,  0.00391164, -0.01180311,
        -0.02494093, -0.00892079,  0.01...,  2.58702432e-06]]), 'R': array([[ 6.86199839e-07, -1.42279806e-08],
       [-1.42279806e-08,  1.81130866e-07]]), ...}
        min_decimal = (11, 11, 16)
/venv/lib/python3.9/site-packages/scipy/linalg/tests/test_solvers.py:651: in _test_factory
    assert_array_almost_equal(res, np.zeros_like(res), decimal=dec)
E   AssertionError: 
E   Arrays are not almost equal to 16 decimals
E   
E   Mismatched elements: 1 / 64 (1.56%)
E   Max absolute difference: 1.81498408e-16
E   Max relative difference: inf
E    x: array([[-3.3055249008362257e-17, -6.5338003968828592e-18,
E           -4.1559671557032246e-18, -3.2208216061107957e-17,
E            2.9366208764832966e-17, -1.0370436131471213e-17,...
E    y: array([[0., 0., 0., 0., 0., 0., 0., 0.],
E          [0., 0., 0., 0., 0., 0., 0., 0.],
E          [0., 0., 0., 0., 0., 0., 0., 0.],...
        a          = array([[ 0.97905961,  0.15428665, -0.03994193,  0.00391164, -0.01180311,
        -0.02494093, -0.00892079,  0.01433595...     [-0.02851398, -0.00156313, -0.01257725, -0.05334177, -0.03769227,
        -0.00788838, -0.62801576,  0.34907   ]])
        b          = array([[-1.04368719,  1.09316659],
       [ 0.51336209, -0.15143908],
       [ 0.78618282,  0.3888125 ],
       [ 0.90... 0.12416864],
       [ 0.90485159, -0.0090001 ],
       [-0.00863767,  0.55645316],
       [ 0.00708354,  0.6553915 ]])
        case       = (array([[ 0.97905961,  0.15428665, -0.03994193,  0.00391164, -0.01180311,
        -0.02494093, -0.00892079,  0.0143359...891e-06,  2.94712339e-08],
       [ 2.40952333e-08,  4.30614350e-07],
       [ 1.23662598e-08, -6.06921957e-07]]), ...)
        dec        = 16
        e          = array([[1., 0., 0., 0., 0., 0., 0., 0.],
       [0., 1., 0., 0., 0., 0., 0., 0.],
       [0., 0., 1., 0., 0., 0., 0., ...   [0., 0., 0., 0., 0., 1., 0., 0.],
       [0., 0., 0., 0., 0., 0., 1., 0.],
       [0., 0., 0., 0., 0., 0., 0., 1.]])
        knownfailure = None
        q          = array([[ 1.75512507e-06, -4.27225221e-07, -9.34790486e-07,
        -6.10641258e-07,  1.62839572e-06,  2.22620136e-06,
...-7.58710361e-07,
        -9.08221173e-07, -2.90720260e-07, -6.17890194e-08,
        -2.00190655e-06,  2.58702432e-06]])
        r          = array([[ 6.86199839e-07, -1.42279806e-08],
       [-1.42279806e-08,  1.81130866e-07]])
        res        = array([[-3.30552490e-17, -6.53380040e-18, -4.15596716e-18,
        -3.22082161e-17,  2.93662088e-17, -1.03704361e-17,
... 2.37834543e-17,
         3.54962391e-18,  2.72929633e-17, -1.01407909e-17,
         7.16701046e-17,  2.62781913e-17]])
        s          = array([[-6.27256153e-07,  2.80554657e-07],
       [ 4.27827567e-07,  8.54401686e-08],
       [ 8.76221935e-07,  1.5285....68302891e-06,  2.94712339e-08],
       [ 2.40952333e-08,  4.30614350e-07],
       [ 1.23662598e-08, -6.06921957e-07]])
        x          = array([[ 1.88357055e-06, -3.98902725e-07, -8.97987558e-07,
        -7.18984081e-07,  1.95043873e-06,  2.15596656e-06,
...-7.62588561e-07,
        -8.85735630e-07, -2.59621086e-07, -7.62321103e-08,
        -1.79447605e-06,  3.08567024e-06]])
=============================== warnings summary ===============================
optimize/tests/test__linprog_clean_inputs.py: 2 warnings
optimize/tests/test_linprog.py: 8 warnings
  /venv/lib/python3.9/site-packages/numpy/core/_asarray.py:83: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray
    return array(a, dtype, copy=False, order=order)

spatial/tests/test_distance.py: 14 warnings
  /venv/lib/python3.9/site-packages/scipy/spatial/distance.py:253: DeprecationWarning: 'wminkowski' metric is deprecated and will be removed in SciPy 1.8.0, use 'minkowski' instead.
    kwargs = _validate_kwargs(np.vstack([XA, XB]), mA + mB, n, **kwargs)

spatial/tests/test_distance.py: 14 warnings
  /venv/lib/python3.9/site-packages/scipy/spatial/distance.py:311: DeprecationWarning: 'wminkowski' metric is deprecated and will be removed in SciPy 1.8.0, use 'minkowski' instead.
    kwargs = _validate_kwargs(X, m, n, **kwargs)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
```

</details>

**Comment by charris** [comment]
I don't know if it is related, but manylinux1 32 bit wheels have been failing for a while, so aren't current in the nightlies.

**Comment by tylerjereddy** [comment]
The issue was resolved by bumping from NumPy `1.19.1` to `1.19.3` in the wheels repo 32-bit Linux + Python 3.9 job. I'll close this and if someone feels that this should also work with NumPy `1.19.1` they can reopen.
